src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -1634,6 +1635,11 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
                 'name': vm_size
             }
             sku_profile_vmsizes_list.append(vmsize_obj)
+
+        if skuprofile_rank:
+            for vm_size, rank in zip(sku_profile_vmsizes_list, skuprofile_rank):
+                vm_size['rank'] = rank
```

Comment:
May I ask if the length of sku_profile_vmsizes_list should be equal to the length of skuprofile_rank?
If so, I think it would be a good idea to add a validation for this. If the length does not match, then an error message is prompted to the user to guide them to enter the correct value.